### PR TITLE
AVM2 Bitmap

### PR DESCRIPTION
**Pull request by kmeisthax**
- [x] `Bitmap`
  - [x] AVM2 object storage on display object side
  - [x] Constructor
  - [x] `bitmapData` property
  - [x] `pixelSnapping` property stub
  - [x] `smoothing` property
- [x] `BitmapData`
  - [x] TObject impl
  - ~~(insert methods we can reasonably implement and test right now)~~ BitmapData impl will be a future PR.

### CODE REVIEW

**Comment by kmeisthax** [comment]
I've decided to just PR this without implementing `BitmapData` methods - it's not particularly necessary for `BitmapData` to be fully implemented, just that you can reference bitmap symbols with it. This is enough to place a library Bitmap onto the display list.

Furthermore, this adds the ability to re-associate a `Bitmap` and it's `BitmapData`; which probably will break some movies. So this will need some testing on bitmap-heavy movies. I can't actually test this in our current test harness because `NullRendererBackend` does not store information about registered bitmaps. I think this is an implementation mistake; it made sense with early ActionScript where there wasn't really a way to get feedback out of the backends. However, as we implement more and more API surface, it's far easier to encounter APIs that explicitly take data back out of these backends; data of which the null ones don't provide. That's something for another PR.

File: tests/tests/swfs/avm2/bitmap_constr/Test.as [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+var b1 = new Bitmap();
+
+trace("///b1.bitmapData;");
trace(b1.bitmapData);
+
+trace("///b1.pixelSnapping;");
+trace(b1.pixelSnapping);

+trace("///b1.smoothing;");
+trace(b1.smoothing);
```
Comment (torokati44) [inline tests/tests/swfs/avm2/bitmap_constr/Test.as:15]: These `^M` control sequences look a bit messy, they are in several .as files.

File: tests/tests/swfs/avm2/bitmap_timeline/Test.as [hunk 2]
```diff
@@ -0,0 +1,1594 @@
...
+}
+
+trace("//true / true");
+assert_divide(true, true);
+
+trace("//false / true");
+assert_divide(false, true);
+
+trace("//null / true");
+assert_divide(null, true);
```
Comment (torokati44) [inline tests/tests/swfs/avm2/bitmap_timeline/Test.as:17]: What do any of these tests in this file have to do with Bitmap?
Is this some unfinished copy-paste artifact? :)

Comment (kmeisthax) [reply tests/tests/swfs/avm2/bitmap_timeline/Test.as:17]: Yes it is, and it's not even in the actual SWF

File: tests/tests/swfs/avm2/bitmap_constr/Test.as (hunk 1 above)
Comment (kmeisthax) [reply tests/tests/swfs/avm2/bitmap_constr/Test.as]: I would really like to know why Animate CC started doing this.

File: core/src/avm2/globals/flash/display/bitmapdata.rs [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                } else {
+                    log::warn!(
+                        "Could not read bitmap data associated with class {:?}",
+                        name
+                    );
+                }
+            }
+        } else {
+            if character.is_some() {
+                //TODO: Determine if mismatched symbols will still work as a
```
Comment (Herschel) [inline core/src/avm2/globals/flash/display/bitmapdata.rs:74]: Looks like this throws a type coercion error when trying to build the frame:
```
TypeError: Error #1034: Type Coercion failed: cannot convert Pic@51ba239 to flash.display.BitmapData.
	at flash.display::Sprite/constructChildren()
	at flash.display::Sprite()
	at flash.display::MovieClip()
```

**Review by Herschel — APPROVED** [review]
Thank you!

**Comment by torokati44** [comment]
So it seems my review request timed out, sorry about that... 😐

**Pull request merged**